src/cargo/sources/git/utils.rs

```diff
@@ -177,7 +177,9 @@ impl GitDatabase {
             Some(c) => c,
             None => GitCheckout::clone_into(dest, self, rev, cargo_config)?,
         };
-        checkout.update_submodules(cargo_config)?;
+        if !cargo_config.offline() {
+            checkout.update_submodules(cargo_config)?;
+        }
```

Comment:
I think this change should be moved into `update_submodules`.  We should probably just mimic #10717 which also tells the user what is happening

That should probably provide good test examples